spring-cloud-stream-reactive/src/main/java/org/springframework/cloud/stream/reactive/StreamEmitterAnnotationBeanPostProcessor.java

```diff
@@ -0,0 +1,244 @@
+				.beansOfTypeIncludingAncestors(this.applicationContext, StreamListenerParameterAdapter.class);
+		for (StreamListenerParameterAdapter parameterAdapter : parameterAdapterMap.values()) {
+			this.streamListenerParameterAdapters.add(parameterAdapter);
+		}
+		Map<String, StreamListenerResultAdapter> resultAdapterMap = BeanFactoryUtils
+				.beansOfTypeIncludingAncestors(this.applicationContext, StreamListenerResultAdapter.class);
+		this.streamListenerResultAdapters.add(new MessageChannelStreamListenerResultAdapter());
+		for (StreamListenerResultAdapter resultAdapter : resultAdapterMap.values()) {
+			this.streamListenerResultAdapters.add(resultAdapter);
+		}
+	}
+
+	@Override
+	public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
+		return bean;
+	}
+
+	@Override
+	public Object postProcessAfterInitialization(final Object bean, final String beanName) throws BeansException {
+		Class<?> targetClass = AopUtils.isAopProxy(bean) ? AopUtils.getTargetClass(bean) : bean.getClass();
+		ReflectionUtils.doWithMethods(targetClass, method -> {
+			StreamEmitter streamEmitter = AnnotatedElementUtils.findMergedAnnotation(method, StreamEmitter.class);
+			if (streamEmitter != null && !method.isBridge()) {
+				targetMethod = method;
+				targetBean = bean;
+			}
+		});
+		return bean;
+	}
+
+	@Override
+	public void start() {
+		if (!running) {
+			if (targetMethod != null) {
+				Assert.isTrue(targetMethod.getAnnotation(Input.class) == null,
+						StreamEmitterErrorMessages.INPUT_ANNOTATIONS_ARE_NOT_ALLOWED);
+				String methodAnnotatedOutboundName = StreamAnnotationCommonMethodUtils.getOutboundBindingTargetName(targetMethod);
+				int outputAnnotationCount = StreamAnnotationCommonMethodUtils.outputAnnotationCount(targetMethod);
+				validateStreamEmitterMethod(targetMethod, outputAnnotationCount, methodAnnotatedOutboundName);
+				invokeSetupMethodOnToTargetChannel(targetMethod, targetBean, methodAnnotatedOutboundName);
+			}
+			this.running = true;
+		}
+	}
+
+	@SuppressWarnings({"rawtypes", "unchecked"})
+	private void invokeSetupMethodOnToTargetChannel(Method method, Object bean, String outboundName) {
+		Object[] arguments = new Object[method.getParameterTypes().length];
+		for (int parameterIndex = 0; parameterIndex < arguments.length; parameterIndex++) {
+			MethodParameter methodParameter = MethodParameter.forMethodOrConstructor(method, parameterIndex);
+			Class<?> parameterType = methodParameter.getParameterType();
+			Object targetReferenceValue = null;
+			if (methodParameter.hasParameterAnnotation(Output.class)) {
+				targetReferenceValue = AnnotationUtils.getValue(methodParameter.getParameterAnnotation(Output.class));
+			}
+			else if (arguments.length == 1 && StringUtils.hasText(outboundName)) {
+				targetReferenceValue = outboundName;
+			}
+			if (targetReferenceValue != null) {
+				Assert.isInstanceOf(String.class, targetReferenceValue, "Annotation value must be a String");
```

Comment:
Looks like this is redundant:
1. The `outboundName` is `String` by definition
2. The `value` attribute of the `@Output` is `String`, too